### PR TITLE
module-qualified sorts

### PR DESCRIPTION
**Pull request by cos**
I don't expect this to work in the near future... just tracking progress.

### CODE REVIEW

**Comment by cos** [comment]
@radumereuta, this is a step forward. 

Still, your commits have regressed the branch in two ways:
1. The transformers are again in one huge method sharing information. Please break it apart again so that any information sharing is explicit.
2. Many tests that were passing before are now failing: see `org.kframework.parser.concrete2kore.RuleGrammarTest` for example.

**Comment by cos** [comment]
Again, it is nice that the passes use transformers now but, from the tests, it is not clear that it actually works.

**Comment by cos** [comment]
@radumereuta, l-am adus la un numar mic de erori, vezi 84d4ff4b5d00465cf0c57cf47d526d9e52ed9d54, dupa care am facut merge cu master-ul.

Acum sunt ceva mai multe erori dar cred ca ne apropiem. Mare parte din teste au trecut end-to-end.

Daca nu-ti place sa lucrezi pe varianta merged, fa un branch din 84d4ff4b5d00465cf0c57cf47d526d9e52ed9d54 pe care-l vom remergui aici. Totusi, cred ca putem tinti si direct sa mearga pe master. Suntem aproape.

**Comment by cos** [comment]
(apologies to non-Romanian speakers for the above message -- no point in translating it as it's outdated already)

@radumereuta, the merged with master version already has only a few errors. We are very close to making it work. Have a look at the parsing errors and I'll deal with the rest tomorrow.

**Comment by cos** [comment]
also, ktest is twice as slow as before, but I think we should deal with the functional bugs first.

**Comment by andreistefanescu** [comment]
revert this hack if no longer necessary: https://github.com/kframework/k/blob/kore-verification/kernel/src/main/java/org/kframework/parser/concrete2kore/ParserUtils.java#L188

File: k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k [hunk 1]
```diff
@@ -442,19 +442,18 @@ current status of the \textsf{return} cell and then pop it when the
 method returns.  We have to do the same with the current object
 environment, so we group them together in the stack frame.  */
 
-  syntax KItem ::= (Map, K, Bag) // TODO: change Bag into ControlCellFragment once issue #1842 is fixed
+  syntax KItem ::= fstackFrame(Map, K, List, Type, K)
```
Comment (radumereuta) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:445]: Does the owner of the language agree with this change?

Comment (grosu) [reply k-distribution/tutorial/2_languages/2_kool/2_typed/1_dynamic/kool-typed-dynamic.k:445]: I don't like it, but it is OK for now, we need to move on.  This needs to be done properly, with a list cell, when the time comes.

**Comment by cos** [comment]
@radumereuta, the merge conflict is in the area you fixed the cell parsing issue in master. You know best what's in there. Could you do the merge?

**Pull request merged**